src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -211,26 +226,61 @@ public boolean addAll(Collection<? extends Student> collection) {
 
     @Override
     public boolean containsAll(Collection<?> collection) {
-        //Ignore this for homework
-        throw new UnsupportedOperationException();
+        for (Object e : collection)
+            if (!contains(e)) {
+                return false;
+            }
+        return true;
     }
 
     @Override
-    public boolean addAll(int i, Collection<? extends Student> collection) {
-        //Ignore this for homework
-        throw new UnsupportedOperationException();
+    public boolean addAll(int i, final Collection<? extends Student> collection) {
+        final Student[] students = collection.toArray(new Student[collection.size()]);
+        final Student[] afterIndex = removeNullsFromArray(Arrays.copyOfRange(elements, i + 1, elements.length - 1));
```

Comment:
same questions as for remove() regards removeNullsFromArray

Reply:
This method corrects the length of an array, by removing nulls immediately it is like a forced GC ))